                             Sobre o conjunto de dados

Visão geral:
Este conjunto de dados inclui dados para a estimativa dos níveis de obesidade em indivíduos dos países do México, Peru e Colômbia, com base em seus hábitos alimentares e condição física.
Os dados contêm 17 atributos e 2111 registros, os registros são rotulados com a variável de classe NObesity (Nível de obesidade), que permite a classificação dos dados usando os valores de Peso insuficiente, Peso normal, Sobrepeso nível I, Sobrepeso nível II, Obesidade tipo I, Obesidade tipo II e Obesidade tipo III.

                        Detalhes da base de dadosDetalhes dos dados:

Gênero: Gênero
Idade: Idade
Altura: em metros
Peso: em kgs
family_history: Algum membro da família sofreu ou sofre de excesso de peso?
FAVC: Você come alimentos altamente calóricos com frequência?
FCVC: Você costuma comer vegetais nas suas refeições?
NCP: Quantas refeições principais você faz diariamente?
CAEC: Você come alguma coisa entre as refeições?
FUMAÇA: Você fuma?
CH2O: Quanta água você bebe diariamente?
SCC: Você monitora as calorias que ingere diariamente?
FAF: Com que frequência você pratica atividade física?
TER: Quanto tempo você usa dispositivos tecnológicos como celular, videogame, televisão, computador e outros?
CALC: Com que frequência você bebe álcool?
MTRANS: Qual meio de transporte você costuma usar?
Nível de obesidade (coluna alvo): Nível de obesidade


Import das bibliotecas necessárias para a análise

In [1]:
# Import da biblioteca que irá acessar e manipular os dados
import pandas as pd

# Import da biblioteca que irá manipular, exibir e criar gráficos
import matplotlib.pyplot as plt

# Import da bivlioteca que irá possibilitar a criação de vários
# tipos de gráficos (é baseado no matplotlib)
import seaborn as sns

# Import da biblioteca que serve para manipular e
# realizar operações com arrays

import numpy as np

Acessando a base de dados

In [2]:
base_dados = pd.read_csv('Obesity prediction.csv')

Analisando as 10 primeiras linhas

In [3]:
base_dados.head(10)

Gender   Age  Height  Weight family_history FAVC  FCVC  NCP       CAEC  \
0  Female  21.0    1.62    64.0            yes   no   2.0  3.0  Sometimes   
1  Female  21.0    1.52    56.0            yes   no   3.0  3.0  Sometimes   
2    Male  23.0    1.80    77.0            yes   no   2.0  3.0  Sometimes   
3    Male  27.0    1.80    87.0             no   no   3.0  3.0  Sometimes   
4    Male  22.0    1.78    89.8             no   no   2.0  1.0  Sometimes   
5    Male  29.0    1.62    53.0             no  yes   2.0  3.0  Sometimes   
6  Female  23.0    1.50    55.0            yes  yes   3.0  3.0  Sometimes   
7    Male  22.0    1.64    53.0             no   no   2.0  3.0  Sometimes   
8    Male  24.0    1.78    64.0            yes  yes   3.0  3.0  Sometimes   
9    Male  22.0    1.72    68.0            yes  yes   2.0  3.0  Sometimes   

  SMOKE  CH2O  SCC  FAF  TUE        CALC                 MTRANS  \
0    no   2.0   no  0.0  1.0          no  Public_Transportation   
1   yes   3.0  yes  3.0  0.0   Sometimes  Public_Transportation   
2    no   2.0   no  2.0  1.0  Frequently  Public_Transportation   
3    no   2.0   no  2.0  0.0  Frequently                Walking   
4    no   2.0   no  0.0  0.0   Sometimes  Public_Transportation   
5    no   2.0   no  0.0  0.0   Sometimes             Automobile   
6    no   2.0   no  1.0  0.0   Sometimes              Motorbike   
7    no   2.0   no  3.0  0.0   Sometimes  Public_Transportation   
8    no   2.0   no  1.0  1.0  Frequently  Public_Transportation   
9    no   2.0   no  1.0  1.0          no  Public_Transportation   

               Obesity  
0        Normal_Weight  
1        Normal_Weight  
2        Normal_Weight  
3   Overweight_Level_I  
4  Overweight_Level_II  
5        Normal_Weight  
6        Normal_Weight  
7        Normal_Weight  
8        Normal_Weight  
9        Normal_Weight

Para facilitar o trabalho, vamos renomear as colunas da tabela

In [4]:
base_dados.rename(columns={
    
    'Gender': 'Genero',
    'Age': 'Idade',
    'Height': 'Altura',
    'Weight': 'Peso',
    'family_history':'Historico_Familiar',
    'FAVC': 'Frequencia_em_alimentos_caloricos',
    'FCVC': 'Come_vegetais_nas_refeiçoes',
    'NCP': 'Quant_refeiçoes_principais',
    'CAEC':'Come_algo_entre_as_refeiçoes',
    'SMOKE': 'Fuma',
    'CH2O': 'Quant_agua_ingerida_diariamente',
    'SCC': 'Monitora_calorias_diariamente',
    'FAF': 'Frequencia_de_atividade_fisica',
    'TUE': 'Tempo_em_dispositivos_eletronicos',
    'CALC':'Frequencia_que_ingere_alcool',
    'MTRANS': 'Transporte_utilizado',
    'Obesity': 'Nivel_Obesidade'
    
    
}, inplace=True)

Verificando se as colunas foram alteradas

In [5]:
base_dados.columns

Index(['Genero', 'Idade', 'Altura', 'Peso', 'Historico_Familiar',
       'Frequencia_em_alimentos_caloricos', 'Come_vegetais_nas_refeiçoes',
       'Quant_refeiçoes_principais', 'Come_algo_entre_as_refeiçoes', 'Fuma',
       'Quant_agua_ingerida_diariamente', 'Monitora_calorias_diariamente',
       'Frequencia_de_atividade_fisica', 'Tempo_em_dispositivos_eletronicos',
       'Frequencia_que_ingere_alcool', 'Transporte_utilizado',
       'Nivel_Obesidade'],
      dtype='object')

Verificando o tipo de cada tabela

In [6]:
base_dados.dtypes

Genero                                object
Idade                                float64
Altura                               float64
Peso                                 float64
Historico_Familiar                    object
Frequencia_em_alimentos_caloricos     object
Come_vegetais_nas_refeiçoes          float64
Quant_refeiçoes_principais           float64
Come_algo_entre_as_refeiçoes          object
Fuma                                  object
Quant_agua_ingerida_diariamente      float64
Monitora_calorias_diariamente         object
Frequencia_de_atividade_fisica       float64
Tempo_em_dispositivos_eletronicos    float64
Frequencia_que_ingere_alcool          object
Transporte_utilizado                  object
Nivel_Obesidade                       object
dtype: object

Verificando a quantidade de valores nulos

In [7]:
base_dados.isnull().sum()

Genero                               0
Idade                                0
Altura                               0
Peso                                 0
Historico_Familiar                   0
Frequencia_em_alimentos_caloricos    0
Come_vegetais_nas_refeiçoes          0
Quant_refeiçoes_principais           0
Come_algo_entre_as_refeiçoes         0
Fuma                                 0
Quant_agua_ingerida_diariamente      0
Monitora_calorias_diariamente        0
Frequencia_de_atividade_fisica       0
Tempo_em_dispositivos_eletronicos    0
Frequencia_que_ingere_alcool         0
Transporte_utilizado                 0
Nivel_Obesidade                      0
dtype: int64

Verificando a quantidade de linhas e colunas da tabela

In [8]:
base_dados.shape

(2111, 17)

                                     ANÁLISE DESCRITIVA

Medidas de tendência central: Calcular a média, mediana e a moda para variáveis numéricas como idade, altura, peso, quantidade de refeições principais, quantidade de água ingerida diaramente e tempo em dispositivos eletrônicos

Média: média aritmética dos valores da coluna

Moda: Valor que aparece com mais frequência em um conjunto de valores

mediana: Valor que se encontra no centro de um conjunto de valores.

Visualizando novamente os nomes das colunas para facilitar a criação da solução

In [9]:
base_dados.columns

Index(['Genero', 'Idade', 'Altura', 'Peso', 'Historico_Familiar',
       'Frequencia_em_alimentos_caloricos', 'Come_vegetais_nas_refeiçoes',
       'Quant_refeiçoes_principais', 'Come_algo_entre_as_refeiçoes', 'Fuma',
       'Quant_agua_ingerida_diariamente', 'Monitora_calorias_diariamente',
       'Frequencia_de_atividade_fisica', 'Tempo_em_dispositivos_eletronicos',
       'Frequencia_que_ingere_alcool', 'Transporte_utilizado',
       'Nivel_Obesidade'],
      dtype='object')

primeiro, vamos criar uma função para facilitar o nosso processo, já que, com a
criação dela, não precisaremos repetir o processo várias vezes

In [10]:
# A função terá como objetivo mostrar no console a média, a mediana e a moda
# de cada coluna. A função ira receber como parametro o nome da coluna que 
# terá os valores impressos

# Criação da função que ira imprimir os valores
def imprimirValores(coluna):
    
    # impressão dos valores
    
    # Name: Ira mostrar o nome da coluna informada nos colchetes da base de dados
    
    # round: Serve para arredontar os valores. Basicamente, ele analisa se o primeiro
    # valor após a virgula é menor ou maior que 5, se for maior ele arredonta pra cima,
    # se for menor que 5 ele arredonta pra baixo. A função recebe 2 valores o valor e 
    # a quatidade de casas decimais que devem aparecer após a virgula.
    
    # mean: Cálcula a média dos valores
    
    # median: calcula a mediana dos valores
    
    # mode: Cálcula a moda dos valores. Ira retorna 2 valores: o indice da moda 
    # (pois pode haver mais de uma moda) e o valor da moda em si
    
    print("Média dos valores da coluna de ", base_dados[coluna].name,": ", round(base_dados[coluna].mean(), 2))
    
    print("Mediana dos valores da coluna de ", base_dados[coluna].name,": ", round(base_dados[coluna].median(), 2))
    
    print("Moda dos valores da coluna de ", base_dados[coluna].name,": ", round(base_dados[coluna].mode(), 2))

Calculando os valores da idade

In [11]:
# Chamada da função

imprimirValores('Idade')

Média dos valores da coluna de  Idade :  24.31
Mediana dos valores da coluna de  Idade :  22.78
Moda dos valores da coluna de  Idade :  0    18.0
Name: Idade, dtype: float64


Calculando os valores da altura

In [12]:
# Chamada da função

imprimirValores('Altura')

Média dos valores da coluna de  Altura :  1.7
Mediana dos valores da coluna de  Altura :  1.7
Moda dos valores da coluna de  Altura :  0    1.7
Name: Altura, dtype: float64


Calculando os valores da coluna de peso

In [13]:
# Chamada da função

imprimirValores('Peso')

Média dos valores da coluna de  Peso :  86.59
Mediana dos valores da coluna de  Peso :  83.0
Moda dos valores da coluna de  Peso :  0    80.0
Name: Peso, dtype: float64


Valores da coluna de quantidade de refeições principais

In [14]:
# chamada do método

imprimirValores('Quant_refeiçoes_principais')

Média dos valores da coluna de  Quant_refeiçoes_principais :  2.69
Mediana dos valores da coluna de  Quant_refeiçoes_principais :  3.0
Moda dos valores da coluna de  Quant_refeiçoes_principais :  0    3.0
Name: Quant_refeiçoes_principais, dtype: float64


Valores da coluna de quantidade de agua ingerida diariamente

In [15]:
# chamada da função

imprimirValores('Quant_agua_ingerida_diariamente')

Média dos valores da coluna de  Quant_agua_ingerida_diariamente :  2.01
Mediana dos valores da coluna de  Quant_agua_ingerida_diariamente :  2.0
Moda dos valores da coluna de  Quant_agua_ingerida_diariamente :  0    2.0
Name: Quant_agua_ingerida_diariamente, dtype: float64


Valores da coluna de quantidade de tempo em dispositivos eletrônicos

In [16]:
# Chamada da função

imprimirValores('Tempo_em_dispositivos_eletronicos')

Média dos valores da coluna de  Tempo_em_dispositivos_eletronicos :  0.66
Mediana dos valores da coluna de  Tempo_em_dispositivos_eletronicos :  0.63
Moda dos valores da coluna de  Tempo_em_dispositivos_eletronicos :  0    0.0
Name: Tempo_em_dispositivos_eletronicos, dtype: float64


Medidas de dispersão: Calcular o desvio padrão e variância para as mesmas
variáveis

Desvio padrão: distância que os valores estão da média 

Variância:  A variância nos mostra a dispersão dos dados em relação à média, mas ela é expressa em unidades ao quadrado. Isso significa que, se os dados originais forem em metros, a variância estará em metros quadrados. Essa unidade pode dificultar a interpretação direta.

Vamos seguir a mesma pratica de anteriormente, vamos criar uma função
para realizar a mesma tarefa quantas vezes for necessária

In [17]:
base_dados.columns

Index(['Genero', 'Idade', 'Altura', 'Peso', 'Historico_Familiar',
       'Frequencia_em_alimentos_caloricos', 'Come_vegetais_nas_refeiçoes',
       'Quant_refeiçoes_principais', 'Come_algo_entre_as_refeiçoes', 'Fuma',
       'Quant_agua_ingerida_diariamente', 'Monitora_calorias_diariamente',
       'Frequencia_de_atividade_fisica', 'Tempo_em_dispositivos_eletronicos',
       'Frequencia_que_ingere_alcool', 'Transporte_utilizado',
       'Nivel_Obesidade'],
      dtype='object')

In [18]:
# Função que irá calcular a variância e o desvio padrão 
# das colunas numéricas
def calc_desvio_variancia(coluna):
    
    # Variável que irá conter a função std da biblioteca numpy 
    # que tem como objetivo calcular o desvio padrão de uma
    # coluna numérica
    desvio_padrao = np.std(base_dados[coluna])
    
    # Variável que irá conter a função var da biblioteca numpy
    # que tem como objetivo calcular a variância de uma coluna
    # numérica.
    variancia = np.var(base_dados[coluna])
    
    # Impressão dos resultados com o round que ira limitar a aparição de casas após a virgula, 
    # no nosso caso queremos apenas 2 casas após a virgula.
    # O name ira capturar apenas o nome da coluna
    print('Desvio padrão da coluna de ', base_dados[coluna].name,": ", round(desvio_padrao, 2))
    print('Variância da coluna de ', base_dados[coluna].name,": ", round(variancia, 2))
    
    

Desvio padrão e variância da coluna de idade

In [19]:
calc_desvio_variancia('Idade')

Desvio padrão da coluna de  Idade :  6.34
Variância da coluna de  Idade :  40.25


Desvio padrão e variância da coluna de altura

In [20]:
calc_desvio_variancia('Altura')

Desvio padrão da coluna de  Altura :  0.09
Variância da coluna de  Altura :  0.01


Desvio padrão e variância da coluna de peso

In [21]:
calc_desvio_variancia('Peso')

Desvio padrão da coluna de  Peso :  26.18
Variância da coluna de  Peso :  685.65


Desvio padrão e variância da coluna de quantidade de refeições principais

In [22]:
calc_desvio_variancia('Quant_refeiçoes_principais')

Desvio padrão da coluna de  Quant_refeiçoes_principais :  0.78
Variância da coluna de  Quant_refeiçoes_principais :  0.61


Desvio padrão e variância da coluna de quantidade de agua ingerida diariamente

In [23]:
calc_desvio_variancia('Quant_agua_ingerida_diariamente')

Desvio padrão da coluna de  Quant_agua_ingerida_diariamente :  0.61
Variância da coluna de  Quant_agua_ingerida_diariamente :  0.38


Variância e desvio padrão da coluna de tempo em dispositivos eletrônicos

In [24]:
calc_desvio_variancia('Tempo_em_dispositivos_eletronicos')

Desvio padrão da coluna de  Tempo_em_dispositivos_eletronicos :  0.61
Variância da coluna de  Tempo_em_dispositivos_eletronicos :  0.37


Frequências: Contar a frequência de cada categoria para variáveis categóricas
como gênero, histórico familiar, se a pessoa fuma ou não e o transporte utilizado.

Definição: A frequência representa a quantidade de vezes que os valores únicos de uma coluna
aparecem, por exemplo, podemos calcular quantas vezes o valor categórico "sim" aparece em uma 
tabela.
Ela pode ser dividida em:

frequencia absoluta: Quantidade de vezes que o valor aparece.

frequência relativa: Porcentagem das vezes que o valor aparece.

Criação da função que irá calcular a frequência de cada valor nas colunas categóricas

In [39]:
# Função que ira calcular a frequência dos valores.
# A função irá receber como parametro o nome da 
# coluna que queremos calcular a frequência.
def calc_frequencia(coluna_categorica):
    
    # Variável que irá conter a função value_counts que tem como objetivo
    # contar a quantidade de vezes um valor aparece na coluna. A função
    # recebe como parametro o normalize=True que tem como objetivo calcular
    # a frequência relativa (porcentagem das vezes que o valor aparece) dos
    # valores da coluna.
    calculo_frequência = base_dados[coluna_categorica].value_counts(normalize=True)
    
    # impressão dos resultados usando o round que ira limitar a aparição de casas decimais
    # após a virgula, no nosso caso iremos limitar a apenas 2 casas decimais
    print('Calculo da frequência dos valores da coluna: ', round(calculo_frequência,2))
    
    

Visualizando as colunas para facilitar a chamada da função

In [27]:
base_dados.columns

Index(['Genero', 'Idade', 'Altura', 'Peso', 'Historico_Familiar',
       'Frequencia_em_alimentos_caloricos', 'Come_vegetais_nas_refeiçoes',
       'Quant_refeiçoes_principais', 'Come_algo_entre_as_refeiçoes', 'Fuma',
       'Quant_agua_ingerida_diariamente', 'Monitora_calorias_diariamente',
       'Frequencia_de_atividade_fisica', 'Tempo_em_dispositivos_eletronicos',
       'Frequencia_que_ingere_alcool', 'Transporte_utilizado',
       'Nivel_Obesidade'],
      dtype='object')

Frequência relativa (porcentagem) dos valores da coluna de generos

In [40]:
calc_frequencia('Genero')

Calculo da frequência dos valores da coluna:  Genero
Male      0.51
Female    0.49
Name: proportion, dtype: float64


frequência relativa (porcentagem) da coluna de históricos familiar

In [41]:
calc_frequencia('Historico_Familiar')

Calculo da frequência dos valores da coluna:  Historico_Familiar
yes    0.82
no     0.18
Name: proportion, dtype: float64


Porcentagem relativa (porcentagem) da coluna Fuma

In [42]:
calc_frequencia('Fuma')

Calculo da frequência dos valores da coluna:  Fuma
no     0.98
yes    0.02
Name: proportion, dtype: float64


Porcentagem relativa da coluna de transportes utilizados

In [44]:
calc_frequencia('Transporte_utilizado')

Calculo da frequência dos valores da coluna:  Transporte_utilizado
Public_Transportation    0.75
Automobile               0.22
Walking                  0.03
Motorbike                0.01
Bike                     0.00
Name: proportion, dtype: float64


Tabelas de Contingência: Criar tabelas cruzadas para analisar a relação entre duas variáveis categóricas, por exemplo, genêro e nivel de obesidade

Primeiro vamos filtrar apenas as colunas categóricas para facilitar a 
visualização das colunas que temos que trabalhar.

Tabelas cruzadas: Uma tabela cruzada, também conhecida como tabulação cruzada ou tabela de contingência, é uma ferramenta estatística que permite comparar os resultados de duas ou mais variáveis categóricas. Ela organiza os dados em uma tabela, cruzando as categorias de cada variável para mostrar a frequência com que ocorrem em conjunto.

In [ ]:
# Vamos usar o a função select_dtypes do pandas que tem como objetivo
# filtrar as colunas por tipo de dados. A função recebe como parametro
# o include = ['tipo do dado que você quer filtrar']
colunas_categoricas = base_dados.select_dtypes(include=['object'])

Verificando as colunas categóricas

In [48]:
colunas_categoricas.columns

Index(['Genero', 'Historico_Familiar', 'Frequencia_em_alimentos_caloricos',
       'Come_algo_entre_as_refeiçoes', 'Fuma', 'Monitora_calorias_diariamente',
       'Frequencia_que_ingere_alcool', 'Transporte_utilizado',
       'Nivel_Obesidade'],
      dtype='object')

Agora, vamos criar uma função que ira construir as tabelas cruzadas entre 2 variáveis
categóricas

Observação: Irei analisar individualmente cada tabela cruzada.

In [52]:
# Função que ira criar as tabelas cruzadas. A função ira receber como parametro o nome das 2 colunas que serão utilizadas na construção da tabela 
# cruzada.

def tabela_cruzada(coluna1, coluna2):
    
    # Ira receber a função do pandas que ira construir a tabela cruzada 
    tabela = pd.crosstab(colunas_categoricas[coluna1], colunas_categoricas[coluna2])
    
    return tabela

Primeiro, vamos fazer a relação entre gênero e nivel de obesidade

Como vamos observar abaixo, para realizarmos uma análise um pouco mais detalhada teremos que pontuar a diferença percentual entre os valores,para
isso, vamos criar uma função que seguira a segunite fórmula matemática
para calculos de diferenças percentuais.

Calculo da porcentagem:

Diferença: valor maior  - valor menor

Dividindo pela base = diferença / valor menor

porcentagem = resulta da divisão pela base * 100 



Criação da função de calculos percentuais

In [62]:
# Função que ira calcular a diferença percentual. A 
# A função irá receber 2 argumentos o valor maior
# e o valor menor
def calc_diferenca_percentual(valormaior, valormenor):
    
    diferenca = valormaior - valormenor
    
    divisao_pela_base = diferenca / valormenor
    
    porcentagem = divisao_pela_base * 100
    
    # Retorno do valor que queremos descobrir
    return porcentagem
    

In [53]:
tabela_cruzada('Genero', 'Nivel_Obesidade')

Nivel_Obesidade  Insufficient_Weight  Normal_Weight  Obesity_Type_I  \
Genero                                                                
Female                           173            141             156   
Male                              99            146             195   

Nivel_Obesidade  Obesity_Type_II  Obesity_Type_III  Overweight_Level_I  \
Genero                                                                   
Female                         2               323                 145   
Male                         295                 1                 145   

Nivel_Obesidade  Overweight_Level_II  
Genero                                
Female                           103  
Male                             187

In [68]:
# Chamada da função de cálculos de porcentagem

calc_diferenca_percentual(187, 103)

81.55339805825243

Conclusão da tabela: Analisando a tabela, percebemos que no geral os homens sofrem com o peso mais do
que as mulheres, entretanto, os casos em que as mulheres sofrem mais, o valor é muito discrepante,
sendo muito maior que os dos homens, como no caso da obesidade 3 por exemplo.

Peso insuficiente: O grupo das mulheres com peso insuficiente é 74% maior que o grupo de homens com
peso insuficiente

peso normal: O grupo de homens com peso normal é 3.54% maior que o grupo de mulheres com
peso normal

obesidade tipo 1:  O grupo de homens com obesidade tipo 1 é 25% maior que o grupo de mulheres com
obesidade tipo 1

Obesidade tipo 2:  O grupo de homens com obesidade tipo 2 é 14650% maior que o grupo de mulheres com
obesidade tipo 2.

Obesidade tipo 3:  O grupo das mulheres com obesidade tipo 3 é 32200% maior que o grupo de homens com
obesidade tipo 3

sobrepeso nivel 1: não há diferença entre os valores

sobrepeso nivel 2:  O grupo dos homens com sobrepeso nivel 2  é 81.53% maior que o grupo das mulheres com
sobrepeso nivel 2

validação entre o histórico familiar e a frequencia de calorias consumidas

In [55]:
tabela_cruzada('Historico_Familiar', 'Frequencia_em_alimentos_caloricos')

Frequencia_em_alimentos_caloricos   no   yes
Historico_Familiar                          
no                                  99   286
yes                                146  1580

In [69]:
# Chamada da função de calculo de diferenças percentuais

calc_diferenca_percentual(1580, 286)

452.4475524475524

Conclusão da tabela: podemos observar que pessoas com histórico familiar 
consomem 452% mais alimentos calóricos que as pessoas sem histórico familiar.

Relação entre a frequência de consumo de alimentos calóricos e o 
nivel de obesidade

In [57]:
tabela_cruzada('Frequencia_em_alimentos_caloricos', 'Nivel_Obesidade')

Nivel_Obesidade                    Insufficient_Weight  Normal_Weight  \
Frequencia_em_alimentos_caloricos                                       
no                                                  51             79   
yes                                                221            208   

Nivel_Obesidade                    Obesity_Type_I  Obesity_Type_II  \
Frequencia_em_alimentos_caloricos                                    
no                                             11                7   
yes                                           340              290   

Nivel_Obesidade                    Obesity_Type_III  Overweight_Level_I  \
Frequencia_em_alimentos_caloricos                                         
no                                                1                  22   
yes                                             323                 268   

Nivel_Obesidade                    Overweight_Level_II  
Frequencia_em_alimentos_caloricos                       
no                                                  74  
yes                                                216

In [72]:
# Chamada da função de cálculo de diferenças percentuais

calc_diferenca_percentual(221, 51)

333.33333333333337

Conclusão da tabela: Podemos observar que pessoas com consomem calorias frequentemente possuem um
nivel bem mais alto em todas as categorias do que pessoas que não consomem calorias frequentemente



peso insuficiente: o grupo de pessoas que consomem calorias é 333% maior que o grupo de pessoas que não
consomem calorias frequentemente

peso normal: o grupo de pessoas que consomem calorias é 163% maior que o grupo de pessoas que não
consomem calorias frequentemente.

obesidade tipo 1: o grupo de pessoas que consomem calorias é 2990% maior que o grupo de pessoas que não
consomem calorias frequentemente. 

Obesidade tipo 2: o grupo de pessoas que consomem calorias é 4042% maior que o grupo de pessoas que não
consomem calorias frequentemente.

Obesidade tipo 3: o grupo de pessoas que consomem calorias é 32200% maior que o grupo de pessoas que não
consomem calorias frequentemente.

Sobrepeso nivel 1: o grupo de pessoas que consomem calorias é 1118% maior que o grupo de pessoas que não
consomem calorias frequentemente.

sobrepeso nivel 2: o grupo de pessoas que consomem calorias é 191% maior que o grupo de pessoas que não
consomem calorias frequentemente.